tests/gold_tests/logging/log_retention.test.py

```diff
@@ -0,0 +1,105 @@
+    'proxy.config.http.wait_for_cache': 1,
+
+    'proxy.config.diags.debug.enabled': 1,
+    'proxy.config.diags.debug.tags': 'logspace',
+
+    'proxy.config.log.rolling_enabled': 3,
+    'proxy.config.log.auto_delete_rolled_files': 1,
+
+    # 10 MB is the minimum rolling size.
+    'proxy.config.log.rolling_size_mb': 10,
+    'proxy.config.log.periodic_tasks_interval': 1,
+    # The following configures a 12 MB log cap with a required 2 MB head room.
+    # Thus the rotated log of just over 10 MB should be deleted because it
+    # will not leave enough head room.
+    'proxy.config.log.max_space_mb_headroom': 2,
+    'proxy.config.log.max_space_mb_for_logs': 12,
+})
+
+
+# Configure approximately 5 KB entries.
+ts.Disk.logging_yaml.AddLines(
+    '''
+logging:
+  formats:
+    - name: long
+      format: "{prefix}: %<sssc>"
+  logs:
+    - filename: test_rotation
+      format: long
+'''.format(prefix="0123456789"*500).split("\n")
+)
+
+# Verify from traffic.out that the rotated log file was auto-deleted.
+ts.Streams.stderr = Testers.ContainsExpression(
+        "logical space used.*space is not available",
+        "It was detected that space was not available")
+ts.Streams.stderr += Testers.ContainsExpression(
+        "auto-deleting.*test_rotation.log",
+        "Verify the test log file got deleted")
+ts.Streams.stderr += Testers.ContainsExpression(
+        "The rolled logfile.*was auto-deleted.*bytes were reclaimed",
+        "Verify that space was reclaimed")
+
+# Create and configure microserver.
+server = Test.MakeOriginServer("server")
+request_header = {"headers": "GET / HTTP/1.1\r\nHost: does.not.matter\r\n\r\n",
+                  "timestamp": "1469733493.993", "body": ""}
+response_header = {"headers": "HTTP/1.1 200 OK\r\nConnection: close\r\nCache-control: max-age=85000\r\n\r\n",
+                   "timestamp": "1469733493.993", "body": "xxx"}
+server.addResponse("sessionlog.json", request_header, response_header)
+ts.Disk.remap_config.AddLine(
+    'map http://127.0.0.1:{0} http://127.0.0.1:{1}'.format(ts.Variables.port, server.Variables.Port)
+)
+
+# The first test run starts the required processes.
+tr = Test.AddTestRun()
+tr.Processes.Default.Command = 'curl "http://127.1.1.1:{0}" --verbose ; '.format(
+        ts.Variables.port)
+tr.Processes.Default.ReturnCode = 0
+server.StartBefore(Test.Processes.ts)
+tr.Processes.Default.StartBefore(Test.Processes.server)
+
+tr.StillRunningAfter = ts
+tr.StillRunningAfter = server
+
+# With the following test run, we instigate a log rotation via entries from a
+# few thousand curl requests.
+tr = Test.AddTestRun()
+# At 5K a log entry, we need a lot of curl'd requests to get to the 10 MB roll
+# minimum.
+curl_commands = 'for i in {{1..2500}}; do curl "http://127.1.1.1:{0}" --verbose; done'.format(
+    ts.Variables.port)
+tr.Processes.Default.Command = curl_commands
+tr.Processes.Default.ReturnCode = 0
```

Comment:
I ran this test:
```
wkaras ~/REPOS/TS2
M$ for i in 1 2 ; do \rm not-there ; echo x ; done ; echo $?
rm: not-there: No such file or directory
x
rm: not-there: No such file or directory
x
0
wkaras ~/REPOS/TS2
M$ for i in 1 2 ; do \rm not-there ; done ; echo $?
rm: not-there: No such file or directory
rm: not-there: No such file or directory
1
wkaras ~/REPOS/TS2
M$
```
This indicates that we would only have to worry about the last curl command failing and causing a non-zero return code.  But if Jason is correct that the curl command is flakey, then checking the return code does seem to create some risk of occasional spurious failures of this gold test.